app/lib/server/functions/setUsername.js

```diff
@@ -79,6 +79,7 @@ export const _setUsername = function(userId, u) {
 		Subscriptions.setUserUsernameByUserId(user._id, username);
 		Subscriptions.setNameForDirectRoomsWithOldName(previousUsername, username);
 		LivechatDepartmentAgents.replaceUsernameOfAgentByUserId(user._id, username);
+		Users.setOwnerUsernameByUserId(user._id, username);
```

Comment:
Shouldn't this be under a flag which indicates this is a Service account's username being modified.
Also, to run this function a user is required to have this permission `edit-other-user-inf`??

Reply:
This is called when owner username is modified. The same happens for subscription and room user info. This function will update only those service accounts where the current user is the owner.